subscription-example/index.js

```diff
@@ -0,0 +1,263 @@
+    const protocol = req.socket.encrypted ? 'https' : 'http';
+
+    // perform /sessions call
+    const response = await checkout.sessions({
+      amount: { currency: "EUR", value: 0 }, // zero-auth transaction
+      countryCode: "NL",
+      merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT, // required
+      reference: orderRef, // required: your Payment Reference
+      shopperReference: SHOPPER_REFERENCE,
+      returnUrl: `${protocol}://${host}/checkout?orderRef=${orderRef}`, // set redirect URL required for some payment methods (ie iDEAL)
+      channel: "Web",
+      // recurring payment settings
+      shopperInteraction: "Ecommerce",
+      recurringProcessingModel: "Subscription",
+      enableRecurring: true
+    });
+
+    console.log(response);
+
+    res.json(response);
+
+  } catch (err) {
+    console.error(`Error: ${err.message}, error code: ${err.errorCode}`);
+    res.status(err.statusCode).json(err.message);
+  }
+
+});
+
+
+/* ################# end API ENDPOINTS ###################### */
+
+/* ################# CLIENT SIDE ENDPOINTS ###################### */
+
+// Index (select a demo)
+app.get("/", (req, res) => res.render("index"));
+
+// Cart (continue to checkout)
+app.get("/preview", (req, res) =>
+  res.render("preview", {
+    type: req.query.type,
+  })
+);
+
+// Subscription page (make a payment)
+app.get("/subscription", (req, res) =>
+  res.render("subscription", {
+    type: req.query.type,
+    clientKey: process.env.ADYEN_CLIENT_KEY
+  })
+);
+
+// Admin Panel page
+app.get("/admin", (req, res) =>
+  res.render("admin/index", {
+    data: getAll()
+  })
+);
+
+// Result page
+app.get("/result/:type", (req, res) =>
+  res.render("result", {
+    type: req.params.type,
+  })
+);
+
+// Invoke to make a payment with a token
+app.get("/admin/makepayment/:recurringDetailReference", async (req, res) => {
+
+  console.log("/admin/makepayment/" + req.params.recurringDetailReference);
+
+  let result = "success"
+
+  try {
+    const response = await checkout.payments({
+      amount: { currency: "EUR", value: 1199 },
+      reference: uuid(),
+      shopperInteraction: "ContAuth", // Continuous Authorization
+      recurringProcessingModel: "Subscription",
+      merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT,
+      shopperReference: SHOPPER_REFERENCE,
+      paymentMethod: {
+        storedPaymentMethodId: req.params.recurringDetailReference
+      }
+    });
+    console.log(response);
+
+    if (response.resultCode == "Authorised") {
+      result = "success";
+    } else {
+      result = "error";
+    }
+
+  } catch (err) {
+    console.error(`Error: ${err.message}, error code: ${err.errorCode}`);
+    result = "error"
+  }
+
+  // Make Payment page
+  res.render("admin/makePayment", {
+    type: result,
+    recurringDetailReference: req.params.recurringDetailReference
+  })
+
+});
+
+// Invoke to disable a token
+app.get("/admin/disable/:recurringDetailReference", async (req, res) => {
```

Comment:
Question @gcatanese, do we want to make all calls have the `/api/`-prefix e.g. `/api/admin/disable`

Reply:
`/api` is used for API calls returning JSON, in this case the user is redirected to a different view

Reply:
Makes sense